From 564a859ce389d03a8beb2124d8665a6ad176a310 Mon Sep 17 00:00:00 2001
From: Vicentiu Galanopulo <vicentiu@balena.io>
Date: Thu, 14 Jan 2021 07:51:11 +0100
Subject: [PATCH] Disable internal imx RTC and external i2c RTC

The current hardware setup for the EP does not contain a coin
battery for the internal RTC on the SoM. The system queries the
device created by the RTC and the retrieved date and time value
is used by chronyd. This leads to chronyd using wrong (default)
date and time value from the RTC. To fix this the internal RTC
will be disabled.

On the EP board we don't have an external RTC on i2c, so that
needs to be disabled.

Upstream-status: Inappropriate [configuration]
Signed-off-by: Vicentiu Galanopulo <vicentiu@balena.io>

---
 arch/arm64/boot/dts/compulab/compulab-imx8mq.dtsi | 1 +
 arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi     | 1 +
 2 files changed, 2 insertions(+)

diff --git a/arch/arm64/boot/dts/compulab/compulab-imx8mq.dtsi b/arch/arm64/boot/dts/compulab/compulab-imx8mq.dtsi
index c3de992e9d3a..e9db68b57789 100644
--- a/arch/arm64/boot/dts/compulab/compulab-imx8mq.dtsi
+++ b/arch/arm64/boot/dts/compulab/compulab-imx8mq.dtsi
@@ -532,6 +532,7 @@
 	em3027: rtc@56 {
 		compatible = "emmicro,em3027";
 		reg = <0x56>;
+		status = "disabled";
 	};
 };
 
diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
index 0fc42771052b..c7ec0f734ad9 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq.dtsi
@@ -713,6 +713,7 @@
 			offset = <0x34>;
 			interrupts = <GIC_SPI 19 IRQ_TYPE_LEVEL_HIGH>,
 				<GIC_SPI 20 IRQ_TYPE_LEVEL_HIGH>;
+			status = "disabled";
 		};
 
 		snvs_pwrkey: snvs-powerkey {
-- 
2.17.1

